test/e2e/status_subresource_test.go

```diff
@@ -46,9 +51,79 @@ func testFinalizerWhenStatusForConfigResourcesEnabled(t *testing.T) {
 
 	p := framework.MakeBasicPrometheus(ns, name, name, 1)
 	pm, err := framework.CreatePrometheusAndWaitUntilReady(ctx, ns, p)
-	require.NoError(t, err, "failed to create Prometheus")
+	require.NoError(t, err)
 	finalizers := pm.GetFinalizers()
 	require.NotEmpty(t, finalizers, "finalizers list should not be empty")
 	err = framework.DeletePrometheusAndWaitUntilGone(ctx, ns, name)
 	require.NoError(t, err, "failed to delete Prometheus with status-cleanup finalizer")
 }
+
+// testServiceMonitorStatusSubresource validates ServiceMonitor status updates upon Prometheus selection.
+func testServiceMonitorStatusSubresource(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(ctx, t, testCtx)
+	framework.SetupPrometheusRBAC(ctx, t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []operator.FeatureGateName{operator.StatusForConfigurationResourcesFeature},
+		},
+	)
+	require.NoError(t, err)
+	name := "servicemonitor-status-subresource-test"
+
+	p := framework.MakeBasicPrometheus(ns, name, name, 1)
+
+	_, err = framework.CreatePrometheusAndWaitUntilReady(ctx, ns, p)
+	require.NoError(t, err, "failed to create Prometheus")
+	smon := framework.MakeBasicServiceMonitor(name)
+
+	sm, err := framework.MonClientV1.ServiceMonitors(ns).Create(ctx, smon, v1.CreateOptions{})
+	require.NoError(t, err)
+
+	sm, err = framework.WaitForServiceMonitorStatus(ctx, sm, monitoringv1.ConditionTrue, 1*time.Minute)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(sm.Status.Bindings))
+	require.Equal(t, 1, len(sm.Status.Bindings[0].Conditions))
+	require.Equal(t, p.Name, sm.Status.Bindings[0].Name)
+	require.Equal(t, p.Namespace, sm.Status.Bindings[0].Namespace)
+	require.Equal(t, monitoringv1.PrometheusName, sm.Status.Bindings[0].Resource)
+	require.Equal(t, monitoringv1.ConditionTrue, sm.Status.Bindings[0].Conditions[0].Status)
```

Comment:
Can we encapsulate these checks in `WaitForServiceMonitorStatus`? E.g. we can pass it the bound Prometheus object and it would select the binding that matches the workload.